### PR TITLE
Added Pulse integration with v4

### PR DESCRIPTION
**Pull request by apesant**
This is the integration for Ooyala player v4 using the Pulse ad player.

### CODE REVIEW

**Comment by angieisrani** [comment]
Does this completely take the place of ad_manager_videoplaza.js?

**Comment by apesant** [comment]
Yes.

File: js/ad_manager_pulse.js [hunk 1]
```diff
@@ -0,0 +1,1110 @@
...
+
+    //Pulse Video plugin
+    var PulsePlayerFactory = function() {
+        this.adManager = {};
+        this.name = "PulseVideoTech";
+        this.encodings = [OO.VIDEO.ENCODING.IMA];
+        this.features = [OO.VIDEO.FEATURE.VIDEO_OBJECT_SHARING_TAKE];
+        this.technology = OO.VIDEO.TECHNOLOGY.HTML5 ;
+        // This module defaults to ready because no setup or external loading is required
+        this.ready = true;
```
Comment (angieisrani) [inline js/ad_manager_pulse.js]: we later removed this property so you can remove it if you would like.  It wont' be used

File: js/ad_manager_pulse.js [hunk 2]
```diff
@@ -0,0 +1,1110 @@
...
+         * @method TemplateVideoFactory#create
+         * @param {object} parentContainer The jquery div that should act as the parent for the video element
+         * @param {string} domId The dom id of the video player instance to create
+         * @param {object} ooyalaVideoController A reference to the video controller in the Ooyala player
+         * @param {object} css The css to apply to the video element
+         * @param {string} playerId The unique player identifier of the player creating this instance
+         * @returns {object} A reference to the wrapper for the newly created element
+         */
+        this.create = function(parentContainer, domId, ooyalaVideoController, css, playerId) {
+            var pulseAdManager = pulseAdManagers[playerId];
```
Comment (angieisrani) [inline js/ad_manager_pulse.js]: if possible, please set the domId on the video element

File: js/ad_manager_pulse.js [hunk 3]
```diff
@@ -0,0 +1,1110 @@
...
+            this._requestSettings = {};
+            this._pulseHost = null;
+            this._podStarted = null;
+            this._waitingForContentPause = false;
+            this._isControllingVideo = false;
+            this._isInAdMode = false;
+            this._isWaitingForPrerolls = false;
+            this._session = null;
+            this.ui = null;
+            this._showAdTitle = false;
```
Comment (angieisrani) [inline js/ad_manager_pulse.js]: is there any reason these are all public?

File: js/ad_manager_pulse.js (hunk 1 above)
Comment (apesant) [inline js/ad_manager_pulse.js]: Do you mean the this.ready property?

Comment (angieisrani) [inline js/ad_manager_pulse.js]: yes

File: js/ad_manager_pulse.js (hunk 3 above)
Comment (apesant) [inline js/ad_manager_pulse.js]: No, you're right, I will remove the unnecessary public attributes.

**Comment by angieisrani** [comment]
One more thought: An acceptible filename would also be "pulse.js" since we moved away from the ad_manager_* naming convention.

**Comment by apesant** [comment]
Can this be merged for the release?

**Comment by pilievOoyala** [comment]
I will try and get this merged today, just want to verify that it builds.  In general though, can you guys build some unit tests?

**Comment by apesant** [comment]
Thanks! We have unit tests in a separate repo on our Stash instance.

**Pull request merged**